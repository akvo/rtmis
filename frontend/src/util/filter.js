export const generateAdvanceFilterURL = (advancedFilters, url) => {
  const queryUrlPrefix = url.includes("?") ? "&" : "?";
  const options = advancedFilters
    .map(({ id, label }) => {
      const query = encodeURIComponent(`${id}||${label.toLowerCase()}`);
      return `options=${query}`;
    })
    .join("&");
  url += `${queryUrlPrefix}${options}`;
  return url;
};
