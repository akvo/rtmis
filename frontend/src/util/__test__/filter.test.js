import { generateAdvanceFilterURL } from "../filter";

describe("util/filter", () => {
  test("test if generateAdvanceFilterURL returns the expected value", async () => {
    const url = "jmp/519630048?administration=1&";
    const advancedFilterFakeData = [
      {
        id: 603100002,
        question: "Whether Urban, Peri Urban or Rural",
        value: 13492,
        label: "Rural",
      },
    ];

    const result = generateAdvanceFilterURL(advancedFilterFakeData, url);
    expect(result).toBe(
      "jmp/519630048?administration=1&&options=603100002%7C%7Crural"
    );
    expect(result).toMatchSnapshot();
  });
});
