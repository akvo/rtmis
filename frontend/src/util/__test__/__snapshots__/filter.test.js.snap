// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/filter test if generateAdvanceFilterURL returns the expected value 1`] = `"jmp/519630048?administration=1&&options=603100002%7C%7Crural"`;
